### ISSUE TITLE
org-archive-subtree does not work on subtrees with clock entries

### ISSUE THREAD

**Issue by Peter-Hertz** [syl20bnr/spacemacs#14033]
#### Description :octocat:
org-archive-subtree does not work on items with clock entries, i.e. items with ":LOGBOOK:", and ":END:"

#### Reproduction guide :beetle:
- Start Emacs
- Create an org file and a subtree
- Clock in, clock out. Three lines are created, looking something like this:

:LOGBOOK:
CLOCK: [2020-08-10 Mon 17:49]--[2020-08-10 Mon 18:34] =>  0:45
:END:

- Try to "org-archive-subtree" the subtree (C-c C-x C-s)

*Observed behaviour:* :eyes: :broken_heart:
- The subtree does not disappear from the org file
- spacemacs gives the following error:
indent-line-to: Wrong type argument: wholenump, -1

*Expected behaviour:* :heart: :smile:
- Archiving the entry (adding it to the archive file, removing it from current file)

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0f65b000d)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
‘‘‘elisp
(javascript html python markdown helm auto-completion better-defaults emacs-lisp git markdown
            (org :variables org-enable-reveal-js-support t)
            spell-checking syntax-checking version-control pandoc themes-megapack racket treemacs)
‘‘‘
- System configuration features: XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD JSON PDUMPER GMP

#### Backtrace :paw_prints:
```
Copied: Subtree(s) with 89 characters
indent-line-to: Wrong type argument: wholenump, -1
```

**Comment by Peter-Hertz** [comment]
After some more tests, the subtree appears to be copied to to archive file correctly, but it is not removed from the original file like it should.
A workaround would be to:
- Achive the subtree with C-c C-x C-a
- Remove the subtree manually from the original file

Please note I said "the subtree **appears** to be copied to to archive file correctly", I might be missing something.

**Comment by Peter-Hertz** [comment]
Solved by cloning the develop branch directly, instead of stable then switching to develop.

`git clone -b develop https://github.com/syl20bnr/spacemacs ~/.emacs.d`

**Comment by Peter-Hertz** [comment]
Problem came back

**Comment by carlos-wong** [comment]
> Problem came back

me too

**Comment by duianto** [comment]
This seems to occur when the logbook lines don't have any indentation.

It's not limited to Spacemacs, because it also happens with just Emacs and the `org-plus-contrib` package.

#### With indentation:
```org
* a
** b
  :LOGBOOK:
  CLOCK: [2020-08-10 Mon 17:49]--[2020-08-10 Mon 18:34] => 0:45
  :END:

```

With the cursor on the subheading: `** b`
After pressing: `C-c C-x C-s` (which calls `org-archive-subtree`)

The buffer contents changed to:
```org
* a

```

The `*Messages*` buffer shows:
>(New file)
Copied: Subtree(s) with 89 characters
Saving file c:/Users/username/Temp/14033/org-archive-subtree.org_archive...
Wrote c:/Users/username/Temp/14033/org-archive-subtree.org_archive
Cut: Subtree(s) with 89 characters
Subtree archived in file: c:/Users/username/Temp/14033/org-archive-subtree.org_archive

The file: `org-archive-subtree.org_archive` contains:
```
#    -*- mode: org -*-


Archived entries from file c:/Users/username/Temp/14033/org-archive-subtree.org


* b
  :PROPERTIES:
  :ARCHIVE_TIME: 2020-11-21 Sat 08:12
  :ARCHIVE_FILE: c:/Users/username/Temp/14033/org-archive-subtree.org
  :ARCHIVE_OLPATH: a
  :ARCHIVE_CATEGORY: org-archive-subtree
  :END:
 :LOGBOOK:
 CLOCK: [2020-08-10 Mon 17:49]--[2020-08-10 Mon 18:34] => 0:45
 :END:

```

#### Without indentation:
```org
* a
** b
:LOGBOOK:
CLOCK: [2020-08-10 Mon 17:49]--[2020-08-10 Mon 18:34] => 0:45
:END:

```

With the cursor on the subheading: `** b`
After pressing: `C-c C-x C-s` (which calls `org-archive-subtree`)

The buffer contents doesn't change.

The `*Messages*` buffer shows:
>(New file)
Copied: Subtree(s) with 83 characters
indent-line-to: Wrong type argument: wholenump, -1

**Comment by duianto** [comment]
It's been reported upstream:
org-archive-subtree errors when logbook lines are unindented
https://lists.gnu.org/archive/html/emacs-orgmode/2020-11/msg00438.html

**Comment by Peter-Hertz** [comment]
> It's been reported upstream:
> org-archive-subtree errors when logbook lines are unindented
> https://lists.gnu.org/archive/html/emacs-orgmode/2020-11/msg00438.html

Thanks a lot!

**Comment by Peter-Hertz** [comment]
Does someone have an idea for a workaroud?
For example a way to indent all logbook lines in a buffer.

**Comment by duianto** [comment]
Someone else has reported the same issue upstream.
Archive an task which has un-indented logbook caused error
https://lists.gnu.org/archive/html/emacs-orgmode/2020-12/msg00087.html

A response to that report seems to have tracked down the source.
https://lists.gnu.org/archive/html/emacs-orgmode/2020-12/msg00088.html
>It bisects to 11ef7454a (org.el (org-fixup-indentation): Fix logbook
drawers indentation, 2020-09-07).

So maybe it will be fixed soon.

**Comment by Peter-Hertz** [comment]
> Someone else has reported the same issue upstream.
> Archive an task which has un-indented logbook caused error
> https://lists.gnu.org/archive/html/emacs-orgmode/2020-12/msg00087.html
> 
> A response to that report seems to have tracked down the source.
> https://lists.gnu.org/archive/html/emacs-orgmode/2020-12/msg00088.html
> 
> > It bisects to 11ef7454a (org.el (org-fixup-indentation): Fix logbook
> > drawers indentation, 2020-09-07).
> 
> So maybe it will be fixed soon.

From today December 9th:
https://lists.gnu.org/archive/html/emacs-orgmode/2020-12/msg00105.html
> As mentioned in the earlier thread, the issue should be fixed by 2eb5f0741

Aha! Thanks everyone for your contributions!

**Comment by duianto** [comment]
org-fixup-indentation: Fix regression in logbook adjustment · 2eb5f0741f
https://code.orgmode.org/bzg/org-mode/commit/2eb5f0741fd6cea0f46a1886b8e63e3a291fe968

**Comment by duianto** [comment]
I'm not getting the error message anymore, so the fix seems to have been included in a recent update.

@Peter-Hertz If your able to confirm that it works, then you can close this issue.
